helpers/middleman/url.rb

```diff
@@ -78,12 +78,12 @@ def localize_url(append = '/', locale_id: current_locale_id, base_url: config[:b
     append = strip_index_from_path(append)
 
     # Set no locale, default locale, or unlocalized pages to have no locale path..
-    if !locale_id || locale_id == default_locale_id || is_unlocalized_page?(page)
+    if locale_id.nil? || locale_id == default_locale_id || is_unlocalized_page?(page) || (!is_valid_locale_id_for_base_url?(locale_id, base_url))
       append = '' if append == '/' # Just incase.  No trailing slash for empty appends, like `sharesight.com`, but still for `sharesight.com/xero/`.
       return "#{base_url}#{append}"
     end
 
-    base_url = "#{base_url}/" if base_url
+    base_url += "/" if !base_url.end_with?('/') && !locale_id.start_with?('/')
```

Comment:
👍 I assume.

Reply:
Ye, it was unnoticed because the JS did override & fix all links. 
But this Ruby helper produced `something.com//us`, at times.